eras/shelley/impl/src/Cardano/Ledger/Shelley/Scripts.hs

```diff
@@ -181,6 +181,14 @@ pattern RequireMOf n ms <- (getRequireMOf -> Just (n, ms))
   where
     RequireMOf n ms = mkRequireMOf n ms
 
+{-# COMPLETE
+  RequireSignature
+  , RequireAllOf
+  , RequireAnyOf
+  , RequireMOf ::
+    ShelleyEra
+  #-}
```

Comment:
Since `COMPLETE` pragmas are considered "orphan" when none of the constructors used in its definition are defined in the same module, we can currently only make this work for `NativeScript` to differentiate between `Shelley` and `Allegra` onwards. We cannot differentiate between other `eras` since they do not introduce any new patterns. For `TxCert` we will keep the existing pragma as is, since we cannot specialise the `Shelley` `TxCert` with the pragma and have it apply also to the rest of the `eras` upto `Babbage`.